### PR TITLE
Client Telemetry: Refactors code to create client telemetry config

### PR DESCRIPTION
**Pull request by sourabh1007**
## Description

As per current design of client telemetry, every value is constant. As part of this PR, all those values which has potential to be configurable in future, are put in `ClientTelemetryConfig` file.
It does not mean that, we are going to expose all of them definitely in future.

**Why is this refactoring required?**
It will give us a clear picture of the options we have, to control client telemetry job. 

In this PR, making sure that `ClientTelemetryConfig` is getting initialized only at `DocumentClient` when we are starting the job and flowing wherever these values are required.

## Type of change
- [] This change requires a documentation update

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj [hunk 1]
```diff
@@ -46,7 +46,7 @@
 	</PropertyGroup>
 
 	<ItemGroup>
-		<None Include="..\..\LICENSE" Pack="true" PackagePath="" Visible="false"/>
+		<None Include="..\..\LICENSE" Pack="true" PackagePath="" Visible="false" />
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj:49]: Its in-consistent with above conventions.

File: Microsoft.Azure.Cosmos/src/Telemetry/ClientTelemetryOptions.cs [hunk 2]
```diff
@@ -14,97 +14,80 @@ internal static class ClientTelemetryOptions
...
         // ConversionFactor used in Histogram calculation to maintain precision or to collect data in desired unit
         internal const int HistogramPrecisionFactor = 100;
+        
         internal const double TicksToMsFactor = TimeSpan.TicksPerMillisecond;
         internal const int KbToMbFactor = 1024;
-
         internal const int OneKbToBytes = 1024;
             
+        internal const string DateFormat = "yyyy-MM-ddTHH:mm:ssZ";
+        internal const string EnvPropsClientTelemetrySchedulingInSeconds = "COSMOS.CLIENT_TELEMETRY_SCHEDULING_IN_SECONDS";
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Telemetry/ClientTelemetryOptions.cs:23]: Lets please have a consistent naming for all env variables?

/cc:@kundadebdatta who is also including some ENV variables

File: Microsoft.Azure.Cosmos/src/Telemetry/Models/Configs/ClientTelemetryConfig.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        internal ClientTelemetryServiceConfig clientTelemetryServiceConfig { get; set; }
+        
+        internal ResourceType AllowedResourceTypes { get; set; } = ResourceType.Document;
+        internal List<int> ExcludedStatusCodes { get; set; } = new List<int> { 404, 409, 412 };
+        internal TimeSpan ClientTelemetryProcessorTimeOut { get; set; } = TimeSpan.FromMinutes(5);
+    }
+
+    internal class MetricsPrecision
+    {
+        internal int RequestLatencyPrecision { get; set; } = 4;
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Telemetry/Models/Configs/ClientTelemetryConfig.cs:24]: Unrelated but wondering why 4 decimal places of precision for latency?

File: Microsoft.Azure.Cosmos/src/Telemetry/Models/Configs/ClientTelemetryConfig.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    using Microsoft.Azure.Documents;
+
+    internal class ClientTelemetryConfig
+    {
+        internal MetricsPrecision precisions { get; set; } = new MetricsPrecision();
+        internal NetworkTelemetryConfig networkTelemetryConfig { get; set; }
+        internal ClientTelemetryServiceConfig clientTelemetryServiceConfig { get; set; }
+        
+        internal ResourceType AllowedResourceTypes { get; set; } = ResourceType.Document;
+        internal List<int> ExcludedStatusCodes { get; set; } = new List<int> { 404, 409, 412 };
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Telemetry/Models/Configs/ClientTelemetryConfig.cs:18]: readonly

File: Microsoft.Azure.Cosmos/src/Telemetry/Models/Configs/ClientTelemetryConfig.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//------------------------------------------------------------
+
+namespace Microsoft.Azure.Cosmos.Telemetry.Models
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.Azure.Documents;
+
+    internal class ClientTelemetryConfig
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Telemetry/Models/Configs/ClientTelemetryConfig.cs:11]: How is config instantiated?

File: Microsoft.Azure.Cosmos/src/Telemetry/ClientTelemetryOptions.cs (hunk 2 above)
Comment (kundadebdatta) [reply Microsoft.Azure.Cosmos/src/Telemetry/ClientTelemetryOptions.cs:23]: @sourabh1007 : Most of our environment variables are defined in this `Cosmos.Direct` class: `Constants.EnvironmentVariables`. As a consistent naming strategy, you can re-name it to something similar to: `AZURE_COSMOS_YOUR_CUSTOM_NAME`, for e.g. `AZURE_COSMOS_REPLICA_VALIDATION_ENABLED`.

Comment (sourabh1007) [reply Microsoft.Azure.Cosmos/src/Telemetry/ClientTelemetryOptions.cs:23]: These environment variables are going to remove in future.

File: Microsoft.Azure.Cosmos/src/Telemetry/Models/Configs/ClientTelemetryConfig.cs (hunk 3 above)
Comment (sourabh1007) [reply Microsoft.Azure.Cosmos/src/Telemetry/Models/Configs/ClientTelemetryConfig.cs:23]: it is same as Java. It was decided to have this precision value.

File: Microsoft.Azure.Cosmos/src/Telemetry/Models/Configs/ClientTelemetryConfig.cs (hunk 5 above)
Comment (sourabh1007) [reply Microsoft.Azure.Cosmos/src/Telemetry/Models/Configs/ClientTelemetryConfig.cs:11]: It is instantiated when client telemetry job is started.

**Comment by sourabh1007** [comment]
out of scope as of now. Let's wait for customer's feedback on current implementation.

**Pull request closed without merging**